sale_commission_margin/models/commission_mixin.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2025 Dixmit - Luis Rodríguez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class CommissionMixin(models.AbstractModel):
+    _inherit = "commission.mixin"
+
+    agent_ids = fields.One2many(
+        comodel_name="commission.line.mixin",
+        inverse_name="object_id",
+        string="Agents & commissions",
+        help="Agents/Commissions related to the invoice line.",
+        compute="_compute_agent_ids",
+        readonly=False,
+        store=True,
+        copy=True,
+        precompute=True,
+    )
```

Comment:
You don't need to re-declare whole field:

```suggestion
    # convert existing fields to precompute
    agent_ids = fields.One2many(precompute=True)
```

Reply:
Hi @pedrobaeza, thanks for your review, but I have tried doing it this way and I get the following error. Is that okay anyway?

` warnings.warn(f"precompute attribute doesn't make any sense on non computed field {self}")`

Reply:
OK, then try this, as the warning is not very fine-tuned:

```suggestion
    # convert existing fields to precompute
    agent_ids = fields.One2many(compute="_compute_agent_ids", precompute=True)
```

Reply:
Thanks, Pedro. I have added your suggestion, but I also had to add the store attribute to avoid this warning:
`warnings.warn(f"precompute attribute has no impact on non-stored field {self}").`

On the other hand, I have added a separate commit with a change to make the deduction information clearer by displaying the commission amount before the margin. This way, I prioritized the view, and now it looks as shown in the following image. Additionally, the commission information in the sale order lines is now displayed before the taxes and amount information. Previously, it was the other way around.

![image](https://github.com/user-attachments/assets/90266978-8b82-4616-8ffd-34d629b9a3a0)